src/main/java/org/javarosa/core/model/ItemsetBinding.java

```diff
@@ -22,22 +31,27 @@
 import org.javarosa.core.util.externalizable.ExtWrapTagged;
 import org.javarosa.core.util.externalizable.Externalizable;
 import org.javarosa.core.util.externalizable.PrototypeFactory;
+import org.javarosa.debug.EvaluationResult;
+import org.javarosa.debug.Event;
 import org.javarosa.model.xform.XPathReference;
 import org.javarosa.xpath.XPathConditional;
+import org.javarosa.xpath.XPathException;
 import org.javarosa.xpath.expr.XPathNumericLiteral;
 import org.javarosa.xpath.expr.XPathPathExpr;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class ItemsetBinding implements Externalizable, Localizable {
-    private static final Logger logger = LoggerFactory.getLogger(ItemsetBinding.class);
+    // Temporarily cached filtered list (not serialized)
+    private List<SelectChoice> latestFilteredChoiceList;
+
+    // Values needed to determine whether the cached list should be expired (not serialized)
+    private Map<TreeReference, IAnswerData> latestTriggerValues;
+    private Long latestRandomizeSeed;
```

Comment:
More than once I’ve confused `latestTriggerValues` with `currentTriggerValues`. Perhaps more distinct names would help? Maybe `previous…` or `cached…` instead of `latest…`?

Reply:
Thank you, I like `cached`!